src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -747,30 +730,63 @@ def work_for_vts(vts, ctx):
         self._create_context_jar(ctx)
 
       # Update the products with the latest classes.
-      self._register_vts([ctx])
+      self.register_extra_products_from_contexts([ctx.target], all_compile_contexts)
 
-    jobs = []
-    invalid_target_set = set(invalid_targets)
-    for ivts in invalid_vts:
-      # Invalidated targets are a subset of relevant targets: get the context for this one.
-      compile_target = ivts.target
-      compile_context = compile_contexts[compile_target]
-      invalid_dependencies = self._collect_invalid_compile_dependencies(compile_target,
-                                                                        invalid_target_set)
+    context_for_target = all_compile_contexts[compile_target]
+    compile_context = self.select_runtime_context(context_for_target)
 
-      jobs.append(Job(self.exec_graph_key_for_target(compile_target),
-                      functools.partial(work_for_vts, ivts, compile_context),
-                      [self.exec_graph_key_for_target(target) for target in invalid_dependencies],
-                      self._size_estimator(compile_context.sources),
-                      # If compilation and analysis work succeeds, validate the vts.
-                      # Otherwise, fail it.
-                      on_success=ivts.update,
-                      on_failure=ivts.force_invalidate))
-    return jobs
+    job = Job(self.exec_graph_key_for_target(compile_target),
```

Comment:
If you're going to indent args this way, please move the first one to a new line and then indent it as well.